x/profiles/keeper/keeper_chain_link.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkquery "github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// StoreChainLink stores the given chain link inside the current context.
+// It assumes that the given chain link has already been validated.
+func (k Keeper) StoreChainLink(ctx sdk.Context, link types.ChainLink, destinationAddress string) error {
+	if _, found := k.GetChainLink(ctx, link.ChainConfig.Name, link.Address); found {
+		return fmt.Errorf("chain link already exists")
+	}
+
+	// Check if address has the profile
+	profile, found, err := k.GetProfile(ctx, destinationAddress)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("address does not have any profile")
+	}
+	// Store chain link to the profile
+	profile.ChainsLinks = append(profile.ChainsLinks, link)
+	if err := k.StoreProfile(ctx, profile); err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	key := types.ChainsLinksStoreKey(link.ChainConfig.Name, link.Address)
+	store.Set(key, k.cdc.MustMarshalBinaryBare(&link))
+	return nil
+}
+
+// GetChainLink returns the chain link corresponding to the given address and the given chain name inside the current context.
+func (k Keeper) GetChainLink(ctx sdk.Context, chainName string, address string) (link types.ChainLink, found bool) {
+	store := ctx.KVStore((k.storeKey))
+
+	bz := store.Get(types.ChainsLinksStoreKey(chainName, address))
+	if bz == nil {
+		return types.ChainLink{}, false
+	}
+	k.cdc.MustUnmarshalBinaryBare(bz, &link)
+	return link, true
+}
+
+// GetAllChainsLinks returns a list of all the chains links inside the given context.
+func (k Keeper) GetAllChainsLinks(ctx sdk.Context) []types.ChainLink {
+	var links []types.ChainLink
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.ChainsLinksPrefix)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var link types.ChainLink
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &link)
+		links = append(links, link)
+	}
+	return links
+}
+
+// GetChainLinksWithPagination returns a list of the links which is paginated from all the links
+func (k Keeper) GetChainsLinksWithPagination(ctx sdk.Context, page int, limit int) []types.ChainLink {
```

Comment:
I'm not convinced this is the best way to implement this method. The `page` and `limit` can be provided by the client empty. This is the case when the client used a particular `Key` instead. 

For this reason, I think we should just simply paginate them as it's now done inside the queries, removing this method all together.

Reply:
@RiccardoM This method is for queryChainsLinks function in `querier.go`. It is not gRPC method to query here so the `sdkquery.FilteredPaginate` method can not be used.

`querier.go`:
https://github.com/desmos-labs/desmos/blob/paul/ibc-links/x/profiles/keeper/querier.go#L149

`sdkquery.FilteredPaginate` in `grpc_query.go`:
https://github.com/desmos-labs/desmos/blob/paul/ibc-links/x/profiles/keeper/grpc_query.go#L93

Reply:
As per #447 I think we can safely remove the legacy REST endpoints for queries. Clients should rely only on gRPC ones. So we can remove the entire `client/rest` folder and all the associated keeper methods (like this one)